sdk/internal/recording/recording.go

```diff
@@ -473,11 +473,35 @@ func defaultOptions() *RecordingOptions {
 	}
 }
 
-func (r RecordingOptions) hostScheme() string {
+func (r RecordingOptions) RouteURL(t *testing.T, rawReq *http.Request) {
+	if GetRecordMode() != LiveMode && !IsLiveOnly(t) {
+		originalURLHost := rawReq.URL.Host
+		rawReq.URL.Scheme = r.scheme()
+		rawReq.URL.Host = r.hostAndPort()
+		rawReq.Host = r.hostAndPort()
+
+		rawReq.Header.Set(UpstreamURIHeader, fmt.Sprintf("%v://%v", r.scheme(), originalURLHost))
+		rawReq.Header.Set(ModeHeader, GetRecordMode())
+		rawReq.Header.Set(IDHeader, GetRecordingId(t))
+	}
+}
+
+func (r RecordingOptions) hostAndPort() string {
```

Comment:
I see `net/url` defines `URL.Host` as optionally including a port. That doesn't align with the URL schema but it admits a reasonable argument for us to align with the standard library here:
```suggestion
func (r RecordingOptions) host() string {
```